### ISSUE TITLE
shipment-management/delivery-services nie zwraca metod dla InPostu

### ISSUE THREAD

**Issue by bmusialekcepd** [allegro/allegro-api#8526]
Korzystam z shipment-management/delivery-services do pobieranie dostepnych metod dostawy. 
w zakładace https://allegro.pl.allegrosandbox.pl/moje-allegro/sprzedaz/ustawienia-dostawy/cennik/edycja/ ustawiam opcje dla InPostu 
<img width="1320" alt="Zrzut ekranu 2024-02-23 o 12 16 23" src="https://github.com/allegro/allegro-api/assets/139112632/6f3ca077-f90c-4b49-ac17-c75227b9be57">
Pomimo tego metoda nie zwraca danych dla InPostu

**Comment by PrzemyslawLukanowski** [comment]
Zrzut, który załączyłeś, dotyczy cennika dostawy. Dodałeś metodę, którą teraz uwzględnimy w cenniku, jeśli odpytasz [GET /sale/shipping-rates/{id}](https://developer.allegro.pl/documentation#operation/getShippingRatesSetUsingGET).

Zasób /shipment-management/delivery-services dotyczy natomiast usług dostawy, za pomocą których możesz nadać przesyłki przez Wysyłam z Allegro. Aby Inpost znalazł się wśród zwracanych danych, musisz dodać nową umowę własną InPost przez link https://allegro.pl.allegrosandbox.pl/moje-allegro/sprzedaz/zamowienia/ustawienia. Dane dostępowe do API ShipX musisz otrzymać bezpośrednio od InPost:

<img width="721" alt="Zrzut ekranu 2024-02-23 o 12 42 39" src="https://github.com/allegro/allegro-api/assets/51473661/dacee6e4-db0d-4848-b044-4b87365f14a4">

**Comment by bmusialekcepd** [comment]
jezeli chce dodac numer przesylki wygenerowany w innym systemie(nie korzystam z wysylam z allegro) to metody dostawy powinienem pobrac rowniez z shipment-management/delivery-services?

**Comment by PrzemyslawLukanowski** [comment]
Nie, z tego endpointu korzystaj tylko wtedy, jeśli wysyłasz przesyłki przez Wysyłam z Allegro. W Twoim przypadku odpowiednie będą metody:

- [GET /sale/delivery-methods](https://developer.allegro.pl/documentation#operation/getListOfDeliveryMethodsUsingGET) - aby pobrać listę wszystkich metod dostawy na Allegro,
- [POST /order/checkout-forms/{id}/shipments](https://developer.allegro.pl/documentation/#operation/createOrderShipmentsUsingPOST) - aby dodać nr przesyłki,
- ew. [GET /order/carriers](https://developer.allegro.pl/documentation#operation/getOrdersCarriersUsingGET) aby jednorazowo pobrać carrierId danych przewoźników, po to aby tę wartośc uwzględnić przy dodawaniu numeru przesyłki.

**Comment by bmusialekcepd** [comment]
zeby powiazac dane z [GET /order/carriers] do [GET /sale/delivery-methods]musialbym stowrzyc wlasny mapper? Api allegro nie daje mozliwosci pobrania tych danych bez mapowania po swojej stronie?

**Comment by PrzemyslawLukanowski** [comment]
Nie udostępniamy takiej możliwości, są to dwa różne endpointy. W GET /sale/delivery-methods zwracamy wszystkie dostępne metody dostawy, na podstawie których możesz m.in. stworzyć cennik dostawy, który będzie składał się z różnych metod. Natomiast w GET /order/carriers zwrócimy wyłącznie dane przewoźników, dla których wspieramy śledzenie statusów przesyłek np. dla InPostu będzie to:

```
        {
            "id": "INPOST",
            "name": "InPost"
        },
```

Jeśli numery przesyłek będziesz dodawać za pomocą [POST /order/checkout-forms/{id}/shipments](https://developer.allegro.pl/documentation/#operation/createOrderShipmentsUsingPOST) to ważne, abyś przekazywał poprawne id, tak aby statusy po naszej stronie były aktualizowane na bieżąco.
